Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -399,6 +399,8 @@ def bottle_formula(f, args:)
         keg.find do |file|
           # Set the times for reproducible bottles.
           if file.symlink?
+            # Need to make symlink permissions consistent on macOS and Linux
+            system "chmod", "-h", "0777", file if OS.mac?
```

Comment:
```suggestion
            File.lchmod 0777, file if OS.mac?
```
Maybe? Probs doesn't make a huge difference though.

Reply:
@carlocab Tried that, doesn't error but also doesn't do anything for no apparent reason?

Reply:
Weird; it works for me:
```
❯ ln -s foo bar && stat bar && brew ruby -e 'File.lchmod 0777, "bar"' && stat bar
  File: bar -> foo
  Size: 3               Blocks: 0          IO Block: 4096   symbolic link
Device: 1000008h/16777224d      Inode: 146971067   Links: 1
Access: (0755/lrwxr-xr-x)  Uid: (  501/carlocab)   Gid: (    0/   wheel)
Access: 2021-07-14 17:47:26.859741433 +0800
Modify: 2021-07-14 17:47:26.859741433 +0800
Change: 2021-07-14 17:47:26.859741433 +0800
 Birth: 2021-07-14 17:47:26.859741433 +0800
  File: bar -> foo
  Size: 3               Blocks: 0          IO Block: 4096   symbolic link
Device: 1000008h/16777224d      Inode: 146971067   Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (  501/carlocab)   Gid: (    0/   wheel)
Access: 2021-07-14 17:47:26.859741433 +0800
Modify: 2021-07-14 17:47:26.859741433 +0800
Change: 2021-07-14 17:47:29.464430119 +0800
 Birth: 2021-07-14 17:47:26.859741433 +0800
```

Reply:
@carlocab Can you try to pull this PR locally at some point and replace it and see if the bottle checksum changes as expected? It didn't for me but it could be more than just this.

Reply:
I pulled your PR locally. Whether I use your change or replace the `chmod` call with `File.lchmod`, I get the bottle checksum we want:
```
❯ brew bottle --json --only-json-tab ncc
==> Determining ncc bottle rebuild...
==> Bottling ncc--0.28.6.big_sur.bottle.1.tar.gz...
==> Detecting if ncc--0.28.6.big_sur.bottle.1.tar.gz is relocatable...
./ncc--0.28.6.big_sur.bottle.1.tar.gz
  bottle do
    rebuild 1
    sha256 cellar: :any_skip_relocation, big_sur: "08a7c339b0b3b35d4f7a6973323ba0e11569cf305c278bf7ee29da060c2e2ad6"
  end
```

Reply:
@carlocab Weird! Applied, thanks.